<?xml version="1.0" encoding="UTF-8"?>

<!--
Company: Alteeve's Niche, Inc.
License: GPL v2+
Author:  Madison Kelly <mkelly@alteeve.ca>

This is the AN::Tools master 'words' file.
-->

<words>
	<meta version="3.0.0" languages="en_CA,jp"/>
	<!-- Canadian English -->
	<language name="en_CA" long_name="Canadian English" description="Created by Madison Kelly (mkelly@alteeve.ca) for the AN::Tools suite of perl modules">
		
		<!-- Alert titles -->
		<key name="alert_title_0001">Debug</key>
		<key name="alert_title_0002">Information</key>
		<key name="alert_title_0003">Notice</key>
		<key name="alert_title_0004">Warning</key>
		<key name="alert_title_0005">Critical</key>
		<key name="alert_title_0006">Warning Cleared</key>
		<key name="alert_title_0007">Critical Cleared</key>
		<key name="alert_title_0008">Important</key>	<!-- This is used for 'warning' level alerts that aren't a problem, but important information. -->
		
		<!-- Messages for users (less technical than log entries), though sometimes used for logs, too. -->
		<key name="message_0001">The host name: [#!variable!target!#] does not resolve to an IP address.</key>
		<key name="message_0002">The connection to: [#!variable!target!#] on port: [#!variable!port!#] as the user: [#!variable!user!#] was refused. If you recently booted the target, it may be just about finished booting. It is normal for the connection to be refused for a brief time during the boot process.</key>
		<key name="message_0003">There is no route to: [#!variable!target!#]. Is the machine in the process of booting up or powering off?</key>
		<key name="message_0004">Timed out while waiting for a reply from: [#!variable!target!#]. Is the machine booting up? If so, please wait a minute or two and try again.</key>
		<key name="message_0005">There was an unknown error while connecting to: [#!variable!target!#]. The error was: [#!variable!error!#]</key>
		<key name="message_0006">We were unable to log in to: [#!variable!target!#]. Please check that the password is correct or that passwordless SSH is configured properly.</key>
		<key name="message_0007">An SSH session was successfully opened to: [#!variable!target!#].</key>
		<key name="message_0008">There was a problem establishing an SSH channel to the target: [#!variable!target!#] for shell call: [#!variable!shell_call!#]</key>
		<key name="message_0009">The SSH session to: [#!variable!target!#] was successfully closed.</key>
		<key name="message_0010">The SSH session to: [#!variable!target!#] was closed because 'no_cache' was set and there was an open SSH connection.</key>
		
		<!-- Log entries -->
		<key name="log_0001">Starting: [#!variable!program!#].</key>
		<key name="log_0002">
This is a "test" entry.

It is multiple lines with single quotes ['] and double-quotes (") and here are random brackets{!}.

It also has replacement variables: [#!variable!first!#] and [#!variable!second!#].
		</key>
		<key name="log_0003">This is a test log entry that contains a secret [#!variable!password!#]!</key>
		<key name="log_0004">This is a test log entry at log level 2.</key>
		<key name="log_0005">This is a test log entry at log level 3.</key>
		<key name="log_0006">This is a test log entry at log level 4.</key>
		<key name="log_0007">This is a test critical log entry.</key>
		<key name="log_0008">This is a test error log entry.</key>
		<key name="log_0009">This is a test alert log entry.</key>
		<key name="log_0010">This is a test emergency log entry.</key>
		<key name="log_0011">About to run the shell command: [#!variable!shell_call!#]</key>
		<key name="log_0012">About to read the file: [#!variable!shell_call!#]</key>
		<key name="log_0013">About to write the file: [#!variable!shell_call!#]</key>
		<key name="log_0014">[ Error ] - There was a problem running the shell command: [#!variable!shell_call!#]. The error was: [#!variable!error!#].</key>
		<key name="log_0015">[ Error ] - There was a problem reading the file: [#!variable!shell_call!#]. The error was: [#!variable!error!#].</key>
		<key name="log_0016">[ Error ] - There was a problem writing the file: [#!variable!shell_call!#]. The error was: [#!variable!error!#].</key>
		<key name="log_0017">Output: [#!variable!line!#].</key>
		<key name="log_0018">About to open the directory: [#!variable!directory!#]</key>
		<key name="log_0019">Variables:</key>
		<key name="log_0020"><![CDATA[[ Error ] - The method: [#!variable!method!#] was called but the: [#!variable!parameter!#] parameter was not passed or it is empty.]]></key>
		<key name="log_0021"><![CDATA[[ Error ] - The method Storage->read_file() was asked to read the file: [#!variable!file!#], but that file does not exist.]]></key>
		<key name="log_0022"><![CDATA[[ Error ] - The method Storage->read_file() was asked to read the file: [#!variable!file!#] which exists but can't be read.]]></key>
		<key name="log_0023">Reading: [#!variable!line!#].</key>
		<key name="log_0024"><![CDATA[[ Error ] - No template file passed to Template->get().]]></key>
		<key name="log_0025"><![CDATA[[ Error ] - No requested template file: [#!variable!source!#] does not exist. Is it missing in the active skin?]]></key>
		<key name="log_0026"><![CDATA[[ Error ] - The requested template file: [#!variable!source!#] is not readable. Please check that it is readable by the webserver user: [#!variable!user_name!#]]]></key>
		<key name="log_0027"><![CDATA[[ Error ] - No template name passed to Template->get().]]></key>
		<key name="log_0028">Successfully read the words file: [#!variable!file!#].</key>
		<key name="log_0029"><![CDATA[[ Error ] - Storage->find() failed to find: [#!variable!file!#].]]></key>
		<key name="log_0030"><![CDATA[[ Warning ] - Template->skin() was asked to set the skin: [#!variable!set!#], but the source directory: [#!variable!skin_directory!#] doesn't exist. Ignoring.]]></key>
		<key name="log_0031"><![CDATA[[ Warning ] - Storage->search_directories() was passed the array: [#!variable!array!#], but it wasn't actually an array. Using @INC + path::directories::tools + \$ENV{'PATH'} for the list of directories to search instead.]]></key>
		<key name="log_0032"><![CDATA[[ Warning ] - Words->read()' called without a file name to read.]]></key>
		<key name="log_0033"><![CDATA[[ Warning ] - Words->read()' asked to read: [#!variable!file!#] which was not found.]]></key>
		<key name="log_0034"><![CDATA[[ Warning ] - AN::Tools::Words->read()' asked to read: [#!variable!file!#] which was not readable by: [#!variable!user!#] (uid/euid: [#!variable!uid!#]).]]></key>
		<key name="log_0035"><![CDATA[[ Warning ] - The config file: [#!variable!file!#] appears to have a malformed line: [#!variable!count!#:#!variable!line!#].]]></key>
		<key name="log_0036"><![CDATA[[ Error ] - The method Database->read_variable() was called but both the 'variable_name' and 'variable_uuid' parameters were not passed or both were empty.]]></key>
		<key name="log_0037"><![CDATA[[ Error ] - The method Database->insert_or_update_variables() method was called but both the 'variable_name' and 'variable_uuid' parameters were not passed or both were empty.]]></key>
		<key name="log_0038"><![CDATA[[ Error ] - The method Storage->change_mode() was called without an invalid 'mode' parameter. It should have been three or four digits, but: [#!variable!mode!#] was passed.]]></key>
		<key name="log_0039">The host: [#!variable!host!#] has released its database lock.</key>
		<key name="log_0040"><![CDATA[[ Error ] - The method Storage->write_file() was asked to write the file: [#!variable!file!#] but it already exists and 'overwrite' was not set. Aborting.]]></key>
		<key name="log_0041"><![CDATA[[ Error ] - The method Storage->write_file() was asked to write the file: [#!variable!file!#] but it is not a full path. Aborting.]]></key>
		<key name="log_0042"><![CDATA[[ Error ] - The method Words->string() was asked to process the string: [#!variable!string!#] which has insertion variables, but nothing was passed to the 'variables' parameter.]]></key>
		<key name="log_0043"><![CDATA[[ Error ] - The method System->call() was called but 'shell_call' was not passed or was empty.]]></key>
		<key name="log_0044">The host: [#!variable!host!#] has renewed its database lock.</key>
		<key name="log_0045">The host: [#!variable!host!#] is requesting a database lock.</key>
		<key name="log_0046"><![CDATA[[ Note ] - The method Storage->copy_file() was asked to copy: [#!variable!source!#] to: [#!variable!target!#], but the target already exists and 'overwrite' wasn't specified, skipping.]]></key>
		<key name="log_0047"><![CDATA[[ Error ] - The method Log->level() was passed an invalid log level: [#!variable!set!#]. Only '0', '1', '2', '3' or '4' are valid.]]></key>
		<key name="log_0048"><![CDATA[[ Warning ] - Testing of AN::Tools is beginning. This will generate warnings and alerts and are not a concern.]]></key>
		<key name="log_0049"><![CDATA[[ Warning ] - Testing of AN::Tools is complete.]]></key>
		<key name="log_0050">[ Error ] - There is a local database defined, but it does not appear to exist and we could not initialize the database server. Is 'postgresql-server' installed?</key>
		<key name="log_0051"><![CDATA[[ Error ] - The method Storage->change_owner() was asked to change the ownership of: [#!variable!target!#] which doesn't exist.]]></key>
		<key name="log_0052"><![CDATA[[ Error ] - The method Storage->copy_file() was called but the source file: [#!variable!source!#] doesn't exist.]]></key>
		<key name="log_0053"><![CDATA[[ Error ] - The 'Database->connect()' method tried to connect to the same database twice: [#!variable!target!#].]]></key>
		<key name="log_0054">
Connecting to Database with configuration ID: [#!variable!id!#]
- driver: .......... [#!variable!driver!#]
- host: ............ [#!variable!host!#]
- port: ............ [#!variable!port!#]
- postgres_password: [#!variable!postgres_password!#]
- name: ............ [#!variable!name!#]
- user: ............ [#!variable!user!#]
- password: ........ [#!variable!password!#]
		</key>
		<key name="log_0055">Initialized PostgreSQL.</key>
		<key name="log_0056">Updated: [#!variable!file!#] to listen on all interfaces.</key>
		<key name="log_0057">Updated: [#!variable!file!#] to require passwords for access.</key>
		<key name="log_0058"><![CDATA[[ Error ] - The method System->remote_call() was called but the port: [#!variable!port!#] is invalid. It must be a digit between '1' and '65535'.]]></key>
		<key name="log_0059">Started the PostgreSQL database server.</key>
		<key name="log_0060">Database user: [#!variable!user!#] already exists with ID: [#!variable!id!#].</key>
		<key name="log_0061"><![CDATA[[ Error ] - The method Get->users_home() was asked to find the home directory for the user: [#!variable!user!#], but was unable to do so.]]></key>
		<key name="log_0062">SSH session opened without a password to: [#!variable!target!#].</key>
		<key name="log_0063">The database with the ID: [#!variable!id!#] did not respond to pings and 'database::#!variable!id!#::ping_before_connect' is not set to '0' in '#!data!path::configs::striker.conf!#', skipping it.</key>
		<key name="log_0064">[ Warning ] - The database: [#!variable!name!#] on host: [#!variable!host!#] with ID: [#!variable!id!#] can not be used, skipping it.</key>
		<key name="log_0065">
The database connection error was:
----------
#!variable!dbi_error!#
----------
		</key>
		<key name="log_0066">Is the database server running on: [#!variable!target!#] and does the target's firewall allow connections on TCP port: [#!variable!port!#]?</key>
		<key name="log_0067"><![CDATA[The password was not passed, and it is required. Please set: [database::#!variable!id!#::password = <password>] in: [#!data!path::configs::striker.conf!#].]]></key>
		<key name="log_0068"><![CDATA[The database user name: [#!variable!user!#] on the host: [#!variable!host!#] is either not the owner of the database: [#!variable!name!#], or the password for that user is incorrect. 
- Please edit: [#!data!path::configs::striker.conf!#]
  * If the user name is not correct, please update: 
    database::#!variable!id!#::user = <user>
  * If the user name is correct, please update: 
    database::#!variable!id!#::password = <password>
		]]></key>
		<key name="log_0069">The connection to the database: [#!variable!name!#] on host: [#!variable!host!#:#!variable!port!#] was refused. Is the database server running?</key>
		<key name="log_0070">The connection to the database: [#!variable!name!#] on host: [#!variable!host!#:#!variable!port!#] failed because the name could not be translated to an IP address. Is this database server's host name in '/etc/hosts'?</key>
		<key name="log_0071">Successfully Connected to the database: [#!variable!name!#] (id: [#!variable!id!#]) on host: [#!variable!host!#:#!variable!port!#].</key>
		<key name="log_0072"><![CDATA[[ Error ] - The method Database->query() was called without a database ID to query and 'sys::read_db_id' doesn't contain a database ID, either. Are any databases available?]]></key>
		<key name="log_0073"><![CDATA[[ Error ] - The method Database->query() was asked to query the database with ID: [#!variable!id!#] but there is no file handle open to the database. Was the connection lost?]]></key>
		<key name="log_0074">About to query: [#!variable!id!#]:[#!variable!query!#]</key>
		<key name="log_0075"><![CDATA[[ Error ] - Failed to prepare the database query: [#!variable!query!#] on: [#!variable!server!#]. The error was: [#!variable!db_error!#]. Note that if the query reports '--', the query was listed as containing sensitive data and '$an->Log->secure' is not set.]]></key>
		<key name="log_0076"><![CDATA[[ Error ] - Failed to execute the database query: [#!variable!query!#] on: [#!variable!server!#]. The error was: [#!variable!db_error!#]. Note that if the query reports '--', the query was listed as containing sensitive data and '$an->Log->secure' is not set.]]></key>
		<key name="log_0077"><![CDATA[[ Error ] - The method Database->initialize() was called without a database ID to query and 'sys::read_db_id' doesn't contain a database ID, either. Are any databases available?]]></key>
		<key name="log_0078"><![CDATA[[ Error ] - The method Database->initialize() was asked to query the database with ID: [#!variable!id!#] but there is no file handle open to the database. Was the connection lost?]]></key>
		<key name="log_0079"><![CDATA[[ Error ] - The method Database->initialize() was asked to initialize the database: [#!variable!server!#] (id: [#!variable!id!#]) but a core SQL file to load wasn't passed, and the 'database::#!variable!id!#::core_sql' variable isn't set. Unable to initialize without the core SQL file.]]></key>
		<key name="log_0080"><![CDATA[[ Error ] - The method Database->initialize() was asked to initialize the database: [#!variable!server!#] (id: [#!variable!id!#]) but the core SQL file: [#!variable!sql_file!#] doesn't exist.]]></key>
		<key name="log_0081"><![CDATA[[ Error ] - The method Database->initialize() was asked to initialize the database: [#!variable!server!#] (id: [#!variable!id!#]) but the core SQL file: [#!variable!sql_file!#] exist, but can't be read.]]></key>
		<key name="log_0082">The database: [#!variable!server!#] needs to be initialized using: [#!variable!sql_file!#].</key>
		<key name="log_0083"></key>
		<key name="log_0084"><![CDATA[[ Error ] - The method Database->query() was asked to query the database: [#!variable!server!#] but no query was given.]]></key>
		<key name="log_0085"><![CDATA[[ Error ] - The method Database->write() was asked to write to the database: [#!variable!server!#] but no query was given.]]></key>
		<key name="log_0086"><![CDATA[[ Error ] - The method System->check_memory() was called without a program name or PID to check.]]></key>
		<key name="log_0087">Testing access to the the database: [#!variable!server!#] prior to query or write. Program will exit if it fails.</key>
		<key name="log_0088">Access confirmed.</key>
		<key name="log_0089"><![CDATA[[ Error ] - The method Database->write() was asked to write to the database with ID: [#!variable!id!#] but there is no file handle open to the database. Was the connection lost?]]></key>
		<key name="log_0090"><![CDATA[[ Error ] - Failed to 'do' the database query: [#!variable!query!#] on: [#!variable!server!#]. The error was: [#!variable!db_error!#]. Note that if the query reports '--', the query was listed as containing sensitive data and '$an->Log->secure' is not set.]]></key>
		<key name="log_0091">Failed to connect to any database.</key>
		<key name="log_0092"><![CDATA[[ Error ] - Unable to connect to the database: [#!variable!server!#] (id: [#!variable!id!#]_.]]></key>
		<key name="log_0093"><![CDATA[[ Error ] - The method Alert->check_alert_sent() was called but the 'modified_date' parameter was not passed and/or 'sys::db_timestamp' is not set. Did the program fail to connect to any databases?]]></key>
		<key name="log_0094">[ Error ] - Failed to start the Postgres server. Please check the system logs for details.</key>
		<key name="log_0095">The database user: [#!variable!user!#] was created with ID: [#!variable!id!#].</key>
		<key name="log_0096">[ Error ] - Failed to add the database user: [#!variable!user!#]! Unable to proceed.</key>
		<key name="log_0097"><![CDATA[[ Error ] - The method Alert->check_alert_sent() was called but the 'set' parameter was not passed or it is empty. It should be 'set' or 'clear'.]]></key>
		<key name="log_0098">
[ Warning ] - Failed to set an alert because this host is not yet in the database. This can happen if the alert was set before this host was added to the database.
* Details of the alert:
- Type:           [#!variable!type!#]
- Sent By:        [#!variable!alert_sent_by!#]
- Record Locator: [#!variable!alert_record_locator!#]
- Name:           [#!variable!alert_name!#]
- Timestamp:      [#!variable!modified_date!#]
		</key>
		<key name="log_0099">[ Error ] - There is no ScanCore database user set for the local machine. Please check: [#!data!path::config::striker.conf!#]'s DB entry: [#!variable!id!#].</key>
		<key name="log_0100">Database user: [#!variable!user!#] password has been set/updated.</key>
		<key name="log_0101"><![CDATA[[ Error ] - The method Alert->register_alert() was called but the 'title_key' parameter was not passed or it is empty and 'header' is enable (default).]]></key>
		<key name="log_0102">I am not recording the alert with message_key: [#!variable!message_key!#] to the database because its log level was lower than any recipients.</key>
		<key name="log_0103">The local machine's UUID was not read properly. It should be stored in: [#!data!sys::host_uuid!#] and contain hexadecimal characters in the format: '012345-6789-abcd-ef01-23456789abcd' and usually matches the output of 'dmidecode --string system-uuid'. If this file exists and if there is a string in the file, please verify that it is structured correctly.</key>
		<key name="log_0104">The database with ID: [#!variable!id!#] for: [#!variable!file!#] is behind.</key>
		<key name="log_0105">ScanCore database: [#!variable!database!#] already exists.</key>
		<key name="log_0106">The database with ID: [#!variable!id!#] for: [#!variable!file!#] and table: [#!variable!table!#] is behind.</key>
		<key name="log_0107">[ Warning ] - Failed to delete the temporary postgres password.</key>
		<key name="log_0108"><![CDATA[[ Error ] - The method Database->insert_or_update_states() was called but the 'state_host_uuid' parameter was not passed or it is empty. Normally this is set to 'sys::data_uuid'.]]></key>
		<key name="log_0109">[ Error ] - Failed to create the ScanCore database: [#!variable!database!#]</key>
		<key name="log_0110">ScanCore database: [#!variable!database!#] created.</key>
		<key name="log_0111">[ Warning ] - Failed to reload the Postgres server. Please check the system logs for details. The updated configuration is probably not active yet.</key>
		<key name="log_0112">Reloaded the PostgreSQL database server.</key>
		<key name="log_0113"><![CDATA[[ Note ] - The 'Database->configure_pgsql() method was called but the parent program is not running with root priviledges. Returning without doing anything.]]>.</key>
		<key name="log_0114"><![CDATA[The program: [#!data!path::exe::an-report-memory!#] must be called with '--program <name>', but no program name was read in.]]></key>
		<key name="log_0115">#!variable!program!# has started.</key>
		<key name="log_0116"><![CDATA[The method: [#!variable!method!#] was passed the parameter: [#!variable!parameter!#] with the value: [#!variable!value!#], which is invalid.]]></key>
		<key name="log_0117"><![CDATA[The method 'Convert->human_readable_to_bytes()' was passed the byte size: [#!variable!size!#] in the string: [sign: #!variable!sign!#, size: #!variable!size!#, type: #!variable!type!#] contains an illegal value. Sizes can only be integers or real numbers. It may also have commas in it which will be removed automatically.]]></key>
		<key name="log_0118"><![CDATA[The method 'Convert->human_readable_to_bytes()' was passed the byte size: [#!variable!size!#] in the string: [sign: #!variable!sign!#, size: #!variable!size!#, type: #!variable!type!#] appears to be a byte size already but the size does not seem to be an integer. Byte sizes can only be signed integers. It may also have commas in it which will be removed automatically.]]></key>
		<key name="log_0119"><![CDATA[The method 'Convert->human_readable_to_bytes()' method was called with the value: [#!variable!value!#] which we split into the size: [#!variable!size!#] and type: [#!variable!type!#]. The type appears to be invalid.]]></key>
		<key name="log_0120"><![CDATA[The method 'Convert->round()' was passed the number: [#!variable!number!#] which contains an illegal value. Only digits and one decimal place are allowed.]]></key>
		<key name="log_0121">Current memory used by: [#!variable!program_name!#] is approximately: [#!variable!bytes!#] bytes (#!variable!hr_size!#).</key>
		<key name="log_0122">The 'smaps' proc file for the process ID: [#!variable!pid!#] was not found. Did the program just close?</key>
		<key name="log_0123"><![CDATA[[ Error ] - Asked 'pgrep' to return the PIDs (process IDs) of the program: [#!variable!program!#] and a non-digit value was returned in: [#!variable!line!#]. This should not have happened.]]></key>
		
		<!-- Test words. Do NOT change unless you update 't/Words.t' or tests will needlessly fail. -->
		<key name="t_0000">Test</key>
		<key name="t_0001">Test replace: [#!variable!test!#].</key>
		<key name="t_0002">Test Out of order: [#!variable!second!#] replace: [#!variable!first!#].</key>
		<!-- Do not use 't_0003'. It is used to test failures caused by calling a non-existent key. -->
		<key name="t_0004">#!FREE!#</key>
		<key name="t_0005">
This is a multi-line test string with various items to insert.

It also has some #!invalid!# replacement #!keys!# to test the escaping and restoring.

Here is the default output language: [#!data!defaults::language::output!#]
Here we will inject 't_0000': [#!string!t_0001!#] 
Here we will inject 't_0002' with its embedded variables: [#!string!t_0002!#]
Here we will inject 't_0006', which injects 't_0001' which has a variable: [#!string!t_0006!#].
		</key>
		<key name="t_0006">This string embeds 't_0001': [#!string!t_0001!#]</key>
		
	</language>
	<!-- 日本語 -->
	<language name="jp" long_name="日本語" description=">Created by Madison Kelly (mkelly@alteeve.ca) for the AN::Tools suite of perl modules.">
		
		<!-- Test words. Do NOT change unless you update 't/Words.t' or tests will needlessly fail. -->
		<key name="t_0000">テスト</key>
		<key name="t_0001">テスト いれかえる: [#!variable!test!#]。</key>
		<key name="t_0002">テスト、 整理: [#!variable!second!#]/[#!variable!first!#]。</key>
		<!-- Do not use 't_0003'. It is used to test failures caused by calling a non-existent key. -->
		<key name="t_0004">#!FREE!#</key>
		<key name="t_0005">
これは、挿入するさまざまな項目を含む複数行のテスト文字列です。

#!無効!#な置換#!キー!#を使ってエスケープとリストアをテストすることもできます。

デフォルトの出力言語は次のとおりです：「#!data!defaults::language::output!#」
ここで、「t_0000」を挿入します：[#!string!t_0001!#]
ここでは、 「t_0002」に埋め込み変数を挿入します：「#!string!t_0002!#」
ここでは変数 「#!string!t_0006!#」を持つ 「t_0001」を注入する 「t_0006」を注入します。
		</key>
		<key name="t_0006">この文字列には「t_0001」が埋め込まれています：「#!string!t_0001!#」</key>
	</language>
</words>
